Mathlib/RingTheory/PowerSeries/Basic.lean

```diff
@@ -1832,9 +1836,55 @@ theorem trunc_add (n) (φ ψ : PowerSeries R) : trunc n (φ + ψ) = trunc n φ +
     · rw [zero_add]
 #align power_series.trunc_add PowerSeries.trunc_add
 
+/--
+`R⟦X⟧` is notation for `PowerSeries R`,
+the semiring of formal power series in one variable over a semiring `R`.
+-/
+scoped notation:9000 R "⟦X⟧" => PowerSeries R
+
+theorem trunc_succ (f : R⟦X⟧) (n : ℕ) :
+    trunc n.succ f = trunc n f + Polynomial.monomial n (coeff R n f) := by
+  rw [trunc, Ico_zero_eq_range, sum_range_succ, trunc, Ico_zero_eq_range]
+
+theorem natDegree_trunc_lt (f : R⟦X⟧) (n) : (trunc (n + 1) f).natDegree < n + 1 := by
+  rw [lt_succ_iff, natDegree_le_iff_coeff_eq_zero]
+  intros
+  rw [coeff_trunc]
+  split_ifs with h
+  · rw [lt_succ, ←not_lt] at h
+    contradiction
+  · rfl
+
+@[simp] lemma trunc_zero' {f : R⟦X⟧} : trunc 0 f = 0 := rfl
+
+theorem eval₂_trunc_eq_sum_range {S : Type*} [Semiring S] {s : S} {G : R →+* S} {n} {f : R⟦X⟧} :
+    (trunc n f).eval₂ G s = ∑ i in range n, G (coeff R i f) * s ^ i := by
+  cases n with
+  | zero =>
+    rw [trunc_zero', range_zero, sum_empty, eval₂_zero]
+  | succ n =>
+    have := natDegree_trunc_lt f n
+    rw [eval₂_eq_sum_range' (hn := this)]
+    apply sum_congr rfl
+    intro _ h
+    rw [mem_range] at h
+    congr
+    rw [coeff_trunc, if_pos h]
+
+@[simp] theorem trunc_X {n} : trunc (n + 2) X = (Polynomial.X : R[X]) := by
```

Comment:
```suggestion
@[simp] theorem trunc_X (n) : trunc (n + 2) X = (Polynomial.X : R[X]) := by
```
It might be more useful in practice to prove this for `trunc n X` under the assumption `n >= 2`. Would you mind adding that variant?